src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.hpp

```diff
@@ -77,9 +83,55 @@ class ShenandoahHeuristics : public CHeapObj<mtGC> {
     union {
       size_t _garbage;          // Not used by old-gen heuristics.
       size_t _live_data;        // Only used for old-gen heuristics, which prioritizes retention of _live_data over garbage reclaim
-    } _u;
+    } _region_union;
+#ifdef ASSERT
+    UnionTag _union_tag;
+#endif
   } RegionData;
 
+  static inline void set_RegionData_region_and_garbage(RegionData& region_data, ShenandoahHeapRegion* region, size_t garbage) {
```

Comment:
Can these static functions just be instance methods? call sites would then be:
```C++
region_data->set_region_and_garbage(region, garbage);
```

Reply:
+1

Reply:
very good suggestions.  it looks much better this way.  will make this change momentarily.